static/js/lib/api_test.js

```diff
@@ -169,6 +169,16 @@ describe("api", function() {
           })
         })
       }
+
+      it("dashboard api returns error if status = 503", () => {
```

Comment:
it's really if the error is any code other than 400 or 401, right? might be good to test a few different error codes here, just for completeness' sake.